subsys/bluetooth/host/hci_core.c

```diff
@@ -3638,9 +3638,23 @@ static void hci_vs_init(void)
 static int hci_init(void)
 {
 	int err;
+
+#if defined(CONFIG_BT_HCI_SETUP)
+	struct bt_hci_setup_params setup_params = { 0 };
+#endif
```

Comment:
Can't you just remove this `#endif` and let everything be inside the same `#if defined (CONFIG_BT_HCI_SETUP)` block? As for the "Setting public address not supported by driver" error, wouldn't it be better to give that already when calling `bt_id_create()` rather than here?